tests/test_tensorflow_keras_models.py

```diff
@@ -373,3 +373,25 @@ def test_sequential_model_save_load_without_input_shape(
         np.testing.assert_array_equal(
             loaded_model.predict(data_x), model.predict(data_x)
         )
+
+    def test_preview(self, tmpdir, api, loss, optimizer, metrics):
+        model = (
+            api(num_hidden=1, num_classes=2, input_dim=3)
+            if api != testing_utils.get_small_subclass_mlp
+            else api(num_hidden=1, num_classes=2)
+        )
+
+        tiledb_uri = os.path.join(tmpdir, "model_array")
+
+        # Compiles the model if optimizer is present
+        if optimizer:
+            model.compile(loss=loss, optimizer=optimizer, metrics=[metrics])
+
+        tiledb_model_obj = TensorflowTileDB(uri=tiledb_uri)
+        if model.built:
```

Comment:
Yes there are 4 tests that return value error. The internals are basically `tf` related. Thing is though that the `ValueError` is quite descriptive and that the value error is documented on the function's docstring.
```

def summary(self, line_length=None, positions=None, print_fn=None):
      """Prints a string summary of the network.
    
      Args:
          line_length: Total length of printed lines
              (e.g. set this to adapt the display to different
              terminal window sizes).
          positions: Relative or absolute positions of log elements
              in each line. If not provided,
              defaults to `[.33, .55, .67, 1.]`.
          print_fn: Print function to use. Defaults to `print`.
              It will be called on each line of the summary.
              You can set it to a custom function
              in order to capture the string summary.
    
      Raises:
          ValueError: if `summary()` is called before the model is built.
      """```